### DISCUSSION TITLE
Move *all* website texts to one central place/repo

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [Codeception/Codeception#6063] [Ideas]
*"Where is the text of this web page coming from?"* is one of the most confusing Codeception questions for me ;-) So I'm suggesting to set up a **central repository** for all website content.

* **Module pages** are coming from the module's source code. I see the original advantage of having everything (=code and documentation) in one place. But I think meanwhile the cons outweigh the pros:
    * The php file gets longer than it already is anyway.
    * Prefixing every line with an asterisk (for the php comment) is cumbersome
    * There is no preview
    * ... which both leads to rendering errors (due to syntax errors) on the web page quite often
    * Documentation is seen as a by-product of the code (not as an achievement in itself)

* **Reference** pages are coming from some source code too - see https://github.com/Codeception/codeception.github.com/pull/400#issuecomment-748808907 - and there's not even a hint about that on the page.

* **Guides** are coming from https://github.com/Codeception/codeception.github.com - even @DavertMik confuses this sometimes, see https://github.com/Codeception/codeception.github.com/pull/366#issuecomment-723649798 ;-)

So in total this system is obscure and unwelcoming to new contributors. So I'm suggesting to move everything to https://github.com/Codeception/codeception.github.com

**Comment by Naktibalda** [comment]
I don't like this proposal at all.

It is a sad fact, but documentation is a by-product of the code here.
Most contributors raise pull requests with code change only, and only update documentation when requested to do so.
Making them update documentation in a different repository increases likelihood that it won't happen at all.

The worst case scenario is if someone makes change in [lib-innerbrowser](https://github.com/Codeception/lib-innerbrowser/), then that change must be manually documented in all framework pages.

I would rather move guide files back to codeception/codeception.

P.S. I could use some help with documenting operators in DB module: https://github.com/Codeception/module-db/pull/12

**Comment by ThomasLandauer** [comment]
> The worst case scenario is if someone makes change in lib-innerbrowser, then that change must be manually documented in all framework pages.

Yeah, that's certainly a point I was missing...

Then let's go the opposite route:
* On *each* page, include this yellow alert box as on the framework pages:
  > Module reference is taken from the source code. Help us to improve documentation. Edit module reference
* On the framework pages, on *each function* add a note where the source code lives.

**Comment by Naktibalda** [comment]
> On each page, include this yellow alert box as on the framework pages: 

The only autogenerated pages missing links to source are [Commands](https://codeception.com/docs/reference/Commands), [Stub](https://codeception.com/docs/reference/Stub) and [Mock](https://codeception.com/docs/reference/Mock).
Module pages, [Extensions](https://codeception.com/extensions) and other pages under Util Classes have links to source.
Guides, [Install](https://github.com/Codeception/codeception.github.com/blob/master/install.html) and [Configuration](https://github.com/Codeception/codeception.github.com/blob/master/docs/reference/Configuration.md) are static files and links can be added manually.

Please update static files and I will update autogenerated files.


> On the framework pages, on each function add a note where the source code lives.

I think that this is an overkill and unnecessary links will make documentation harder to read.
There is a link to the source of the module at the top of each module page and the source of all functions are 2 clicks away in IDE.
An absolute majority of our users are using IDE.

Here is a walkthrough if you aren't using IDE:

1. In Cept file I Ctrl-Click'ed on fillField in `$I->fillField('param1', 'foo');` line.
2. It took me to tests/_support/_generated/FunctionalTesterActions.php file where 2 lines above selected line I saw:
     > @see \Codeception\Lib\InnerBrowser::fillField()
3. Ctrl-Click on fillField in that line took me to correct line in InnerBrowser class https://github.com/Codeception/lib-innerbrowser/blob/1.3.4/src/Codeception/Lib/InnerBrowser.php#L1073

When I clicked on addServiceToContainer, comment in FunctionalTesterActions said `@see \Codeception\Module\Laminas::addServiceToContainer()`

How about adding something along the lines of "Majority of actions are inherited from [InnerBrowser](https://github.com/Codeception/lib-innerbrowser/blob/master/src/Codeception/Lib/InnerBrowser.php)" below yellow alert box?

**Reply by ThomasLandauer** [reply]
Atom doesn't have clickable `@see` links :-( https://gitlab.com/Serenata/Serenata/-/issues/344

And due to the Symfony Module being refactored into traits, the link in the yellow alert box at the bottom of https://codeception.com/docs/modules/Symfony is now broken. So I still think a link to the source code above each function would be a good idea. Arguments:

* It increases transparency, and also gives the impression that PR's are welcome.
* What you're saying about IDE is certainly true for code PR's, but I think more of docs PR's here. When people are reading the docs in their browser it should be easy to suggest an improvement.
* Nobody is reading the framework pages from beginning to end; so having some links in there is just the same as the parameters lists that are now in there: If you don't need it, you just skip over it. Besides, those links could be rendered as `<small>`.

This only makes sense if those links could be generated automatically!
